lamindb/_curate.py

```diff
@@ -111,6 +111,15 @@ def __repr__(self) -> str:
 class BaseCurator:
     """Curate a dataset."""
 
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        import sys
+
+        # Deprecated methods
+        if "sphinx" not in sys.modules:
```

Comment:
Arrgh, this seems an incredibly manual hack to account for rendering this in sphinx. Is this always needed as part of the deprecation process?

Reply:
This has **always** been a part of the "Lukas deprecration process". See my original PR where I tried like 5 different solutions till I found this one that worked. Now actually, here we might get a way with a different solution because this isn't a Django model where there's a lot more complexity involved but for consistency, we can go for the same process.

Reply:
Got it! :D

Reply:
Got it! :D

Reply:
I don't think we need to hide it, showing deprecates isn't enough? I don't think we do this for all deprecated methods, do we?

Reply:
No, we hide deprecated methods because we don't want users to see nor use them. They might even not have any functionality like this one. Our API docs are already extremely cluttered especially with some more Django stuff. I'd always hide deprecated methods.